src/views/login.jsx

```diff
@@ -1,7 +1,162 @@
-import React from "react";
+/* eslint-disable react/no-unescaped-entities */
+import { useState } from "react";
+import {
+  signInWithEmailAndPassword,
+  GoogleAuthProvider,
+  signInWithPopup,
+} from "firebase/auth";
+import { auth } from "../../firebase_config";
+import { Link, useNavigate } from "react-router-dom";
 
-const login = () => {
-  return <div>login</div>;
-};
+function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const provider = new GoogleAuthProvider();
+  const navigate = useNavigate();
 
-export default login;
+  const handleLogIn = async (e) => {
+    e.preventDefault();
+
+    signInWithEmailAndPassword(auth, email, password)
+      .then(() => {
+        navigate("/dashboard");
+      })
+      .catch((error) => {
+        if (error.code === "auth/user-not-found") {
+          alert("user not found");
```

Comment:
we may want to consider using a more user-friendly way to display errors, such as a toast notification or an inline error messages. I think we talked about using the npm package toasty for this.